executor/join_test.go

```diff
@@ -1889,3 +1889,20 @@ func (s *testSuiteJoinSerial) TestOuterMatchStatusIssue14742(c *C) {
 		"2 1",
 	))
 }
+
+func (s *testSuiteJoinSerial) TestInlineProjection4HashJoinIssue15316(c *C) {
```

Comment:
This case needn't stay in `testSuiteJoinSerial`

Reply:
updated the test which still needs the `testSuiteJoinSerial`